src/raw/mod.rs

```diff
@@ -3846,6 +3869,41 @@ impl<T> RawIterRange<T> {
             self.next_ctrl = self.next_ctrl.add(Group::WIDTH);
         }
     }
+
+    /// # Safety
+    /// The provided `n` value must match the actual number of items
+    #[allow(clippy::while_let_on_iterator)]
+    #[cfg_attr(feature = "inline-more", inline)]
+    unsafe fn fold_impl<F, B>(mut self, mut n: usize, mut acc: B, mut f: F) -> B
+    where
+        F: FnMut(B, Bucket<T>) -> B,
+    {
+        if n == 0 {
+            return acc;
+        }
+
+        loop {
+            while let Some(index) = self.current_group.next() {
+                debug_assert!(n != 0);
```

Comment:
```suggestion
                debug_assert!(n != 0);
                // The returned `index` will always be in the range `0..Group::WIDTH`,
                // so that calling `self.data.next_n(index)` is safe (see detailed explanation below).
```